sys/parsoid.js

```diff
@@ -430,7 +430,12 @@ PSP.getFormat = function(format, hyper, req) {
                     // Re-try to retrieve from storage with the
                     // normalized title & revision
                     rp.title = revInfo.title;
-                    return self.getFormat(format, hyper, req);
+                    return self.getFormat(format, hyper, req)
+                    .then(function(res) {
+                        // Disable caching for non-canonical requests, as
+                        // those are not purged.
+                        res.headers['cache-control'] = 'no-cache, must-revalidate';
```

Comment:
Functionally, it doesn't really matter. This is a temporary patch, so the idea was to get it out quickly, to avoid non-canonical requests receiving outdated cached content. The entire retry / normalization logic can be removed once we redirect.